pkg/apis/operator/v1/types.go

```diff
@@ -257,6 +257,10 @@ type NodeAddressAutodetection struct {
 	// specified IP or domain.
 	// +optional
 	CanReach string `json:"canReach,omitempty"`
+
+	// CIDRS enables IP auto-detection based on which addresses on the nodes are within
+	// one of the provided CIDRs.
+	CIDRS []string `json:"cidrs,omitempty"`
```

Comment:
Is there anything that can be added here so that there is CIDR validation done on these strings by the apiserver?

Reply:
I don't think so. There's no "validate as cidr" option on CRDs to my knowledge, and regexes for any possible IPv4 / v6 address are _disgusting_. we'll need to do it in code.